### PR TITLE
Fix verbosity bug in --collect-only

### PR DESCRIPTION
**Pull request by thisch**
Closes: #5383

- [ ] Target the `master` branch for bug fixes, documentation updates and trivial changes.
- [ ] Target the `features` branch for new features and removals/deprecations.
- [ ] Include documentation when adding new features.
- [ ] Include new tests or update existing tests when applicable.

Unless your change is trivial or a small documentation fix (e.g.,  a typo or reword of a small section) please:

- [ ] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](https://github.com/pytest-dev/pytest/blob/master/changelog/README.rst) for details.
- [ ] Add yourself to `AUTHORS` in alphabetical order;

### CODE REVIEW

File: src/_pytest/logging.py [hunk 1]
```diff
@@ -628,6 +624,10 @@ def pytest_sessionstart(self):
     @pytest.hookimpl(hookwrapper=True)
     def pytest_runtestloop(self, session):
```
Comment (thisch) [inline src/_pytest/logging.py]: I'm a bit surprised that this hook is called even when --collect-only is used

**Comment by nicoddemus** [comment]
Thanks @thisch for the quick PR!

While it fixes the main problem, using `log-cli-level` still produces the `===` section:

```
λ pytest .tmp\test-foo.py --collect-only -q
test-foo.py::test_post_mortem

no tests ran in 0.01 seconds

 λ pytest .tmp\test-foo.py --collect-only -q --log-cli-level=INFO
test-foo.py::test_post_mortem

=================== no tests ran in 0.01 seconds ====================
```

**Comment by nicoddemus** [comment]
Do we still want to capture log messages during `--collect-only`? If not this patch (over this PR) fixes this:

```diff
diff --git a/src/_pytest/logging.py b/src/_pytest/logging.py
index f8ff0362c..1cffda1c3 100644
--- a/src/_pytest/logging.py
+++ b/src/_pytest/logging.py
@@ -624,16 +624,19 @@ class LoggingPlugin:
     @pytest.hookimpl(hookwrapper=True)
     def pytest_runtestloop(self, session):
         """Runs all collected test items."""
-        if self._log_cli_enabled() and self._config.getoption("verbose") < 1:
-            # setting verbose flag is needed to avoid messy test progress output
-            self._config.option.verbose = 1
-
-        with self.live_logs_context():
-            if self.log_file_handler is not None:
-                with catching_logs(self.log_file_handler, level=self.log_file_level):
+        if session.config.option.collectonly:
+            yield
+        else:
+            if self._log_cli_enabled() and self._config.getoption("verbose") < 1:
+                # setting verbose flag is needed to avoid messy test progress output
+                self._config.option.verbose = 1
+
+            with self.live_logs_context():
+                if self.log_file_handler is not None:
+                    with catching_logs(self.log_file_handler, level=self.log_file_level):
+                        yield  # run all the tests
+                else:
                     yield  # run all the tests
-            else:
-                yield  # run all the tests
```

**Comment by thisch** [comment]
> Do we still want to capture log messages during `--collect-only`? If not this patch (over this PR) fixes this:

I dont understand your comment. live-logging/capturing of the log messages that are created in the collection phase still works, even with your patch.

Note that I've already applied your patch, thx!

**Comment by nicoddemus** [comment]
> I dont understand your comment. live-logging/capturing of the log messages that are created in the collection phase still works, even with your patch.

Sorry, I was not very clear: I meant that my patch fixes the output difference from my previous comment (the `==== no tests run in ... ====` footer).

File: testing/logging/test_reporting.py [hunk 2]
```diff
@@ -916,14 +916,45 @@ def test_collection_live_logging(testdir):
...
+                "*<Module test_collection_collect_only_live_logging.py>*",
+                "*no tests ran*",
+            ]
+        )
+    elif verbose == "-q":
+        assert "collected 1 item*" not in result.stdout.str()
+        expected_lines.extend(
+            [
+                "*test_collection_collect_only_live_logging.py::test_simple*",
+                "*no tests ran*",
```
Comment (nicoddemus) [inline testing/logging/test_reporting.py]:
```suggestion
                "no tests ran in * seconds",
```

This ensures that the `===` footer is not being shown here.

Comment (nicoddemus) [reply testing/logging/test_reporting.py]: Went ahead and applied this myself (and pushed force to your branch, hope you don't mind).

**Comment by nicoddemus** [comment]
Backport: https://github.com/pytest-dev/pytest/pull/5411

**Pull request merged**